### ISSUE TITLE
incorrect use of readBin endian ("little" should be "small")

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1634]
I received an email (private, so not quoted here) pointing this out that `echosounder.R` has an incorrect line
```R
        code <- readBin(buf[offset+3:4], "integer", size=2, n=1, endian="small", signed=FALSE)
```
which ought to be
```R
        code <- readBin(buf[offset+3:4], "integer", size=2, n=1, endian="little", signed=FALSE)
```

The email noted that this will lead to a runtime error in an upcoming R.

**Comment by dankelley** [comment]
FYI, there are no warnings in R-3.5.2 or R-devel (downloaded this morning).  This perhaps suggests that a conversion to a runtime error is more than a few weeks off.

**Comment by richardsc** [comment]
We use readBin in a number of places other than just echosounder. Are they all affected?

**Comment by dankelley** [comment]
No, there was only one instance.  I think it affected only "time" data chunks, and not "beam" chunks, and then only in the second of two varieties of specifying those types.  If you do a diff on the new "develop" commit (e1ebf1ed3528571738c1e36b61c23ea9d2c5e8c2) and the previous one, you can see the changes, the highlight of which is
```diff
-        } else if (code1 == 0x0f || code == 0x20) {
+        } else if (code2 == 0x00 && (code1 == 0x0f || code1 == 0x20)) {
+            ## In [1 table 3.5] these cases are listed as "0x000F or 0x0020 Time"
```

**Comment by dankelley** [comment]
Also, we no longer use 'code', so it is commented out now in "develop" commit  694de6c8ef570c6c384a136826519875203c6b83 (even though, in this commit, the word was changed from `"small"` to `"little"`).

**Comment by dankelley** [comment]
I checked with the original reporter, Tomas Kalibera (@kalibera) to see if it was all right to post the original email, and the answer was "yes", so I am putting that below.  Also, Tomas noted that the general issue of getting `readbin()` to check for improper `endian` values was brought up by Jennifer Lyon  on the R-devel group (https://stat.ethz.ch/pipermail/r-devel/2019-November/078715.html).  I'm grateful to both, for their help in improving R software in general, and `oce` in particular.

> Dear Dan,
>
>it seems that oce is using invalid specification of endian for readBin, oce/R/echosounder.R has
>
>code <- readBin(buf[offset+3:4], "integer", size=2, n=1, endian="small", signed=FALSE)
>
>but the correct values for endian are "big", "little" or "swap". Please note that just changing "small" to "little" will change the behavior for most users. On little endian machines, currently endianness will be swapped, but when you change to "little", it will not be swapped, so please take this into account when deciding how to fix.
>
>Invalid specification of endian will now be turned into a runtime error by R, so your package will start failing, please fix it as soon as possible.
>
>Thanks
>Tomas

**Comment by dankelley** [comment]
Fixed in "develop" branch commit 694de6c8ef570c6c384a136826519875203c6b83 et seq.
